src/geom/face_tri3.C

```diff
@@ -203,4 +203,50 @@ std::pair<Real, Real> Tri3::min_and_max_angle() const
                         std::max(theta0, std::max(theta1,theta2)));
 }
 
+bool Tri3::contains_point (const Point & p, Real tol) const
+{
+  // move test point relative to node 0
+  const Point p0 ( p - this->point(0) );
+
+  // define two in plane vectors
+  Point v1 ( this->point(1) - this->point(0) );
+  Point v2 ( this->point(2) - this->point(0) );
+
+#if LIBMESH_DIM == 3
+  // define out of plane vector
+  Point oop ( v1.cross(v2) );
+
+  // project moved test point onto out of plane component and bail
+  // if it is farther out of plane than tol.
+  if ( std::abs(p0 * oop) > tol || oop.norm_sq() < tol * tol)
```

Comment:
It seems like all we should need is the "barycentric technique" described at `http://blackpawn.com/texts/pointinpoly`... there should not be any cross products, etc. required. I'll try and take a look tonight unless you are already working on that.